pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -380,6 +380,24 @@ func (facts *PipelineRunFacts) GetFinalTasks() PipelineRunState {
 	return tasks
 }
 
+// IsFinalTaskStarted returns true if all DAG pipelineTasks is finished and one or more final tasks have been created.
+func (facts *PipelineRunFacts) IsFinalTaskStarted() bool {
+	// check either pipeline has finished executing all DAG pipelineTasks,
+	// where "finished executing" means succeeded, failed, or skipped.
+	if facts.checkDAGTasksDone() {
+		// return list of tasks with all final tasks
+		for _, t := range facts.State {
+			if facts.isFinalTask(t.PipelineTask.Name) {
+				if len(t.TaskRuns) > 0 || len(t.CustomRuns) > 0 {
+					return true
+				}
```

Comment:
I wonder if this is equal to 
```
t.IsRunning()
```

Reply:
> I wonder if this is equal to
> 
> ```
> t.IsRunning()
> ```

Thanks for your tip, I adjusted and used the t.isScheduled() function to replace it. IsRunning may not be particularly accurate as it excludes the end state.

Reply:
Oh nice, thanks!